lib/api/readable.js

```diff
@@ -146,15 +146,39 @@ module.exports = class BodyReadable extends Readable {
 
   async dump (opts) {
     let limit = opts && Number.isFinite(opts.limit) ? opts.limit : 262144
+    const signal = opts && opts.signal
+    function throwIfAborted () {
```

Comment:
Could we move this into the util file? `throwIfAborted(signal)`

Reply:
Done!